src/compiler/resolutionCache.ts

```diff
@@ -771,17 +769,24 @@ namespace ts {
             }
             const paths = new Set<string>();
             paths.add(locationToWatch);
+            let actualWatcher = canWatchDirectoryOrFile(resolutionHost.toPath(locationToWatch)) ?
+                resolutionHost.watchAffectingFileLocation(locationToWatch, (fileName, eventKind) => {

```

Comment:
Out of curiosity, can this ever produce a `noopFileWatcher`? I was wondering if there's an advantage to producing `noopFileWatcher` instead of `undefined` when `canWatchDirectoryOrFile(...)` is false.

Reply:
Its just easier to not have to check null value before calling close..